From d669f23551329a05b8ddf6eb6f2f0de7ba3ec303 Mon Sep 17 00:00:00 2001
From: Kumar Gala <galak@kernel.crashing.org>
Date: Wed, 17 Feb 2010 20:45:14 -0600
Subject: [PATCH 079/233] powerpc/p3041: Update p3041 DS board support

Extracted from vendor drop QorIQ-DPAA-SDK-V1-20110609-systembuilder.iso
0177-powerpc-p3041-Update-p3041-DS-board-support.patch

Update the P3041 DS boards support for networking and other code
refactoring.

Signed-off-by: Kumar Gala <kumar.gala@freescale.com>
[context adjustment]
Integrated-by: Weiwei Wang <weiwei.wang@windriver.com>
---
 arch/powerpc/platforms/85xx/Kconfig    |    1 +
 arch/powerpc/platforms/85xx/p3041_ds.c |   17 ++++++++++++++++-
 2 files changed, 17 insertions(+), 1 deletions(-)

diff --git a/arch/powerpc/platforms/85xx/Kconfig b/arch/powerpc/platforms/85xx/Kconfig
index 317926a..432e255 100644
--- a/arch/powerpc/platforms/85xx/Kconfig
+++ b/arch/powerpc/platforms/85xx/Kconfig
@@ -161,6 +161,7 @@ config P3041_DS
 	select SWIOTLB
 	select MPC8xxx_GPIO
 	select HAS_RAPIDIO
+	select HAS_FSL_PAMU
 	select FSL_HYDRA_DS_MDIO if PHYLIB
 	help
 	  This option enables support for the P3041 DS board
diff --git a/arch/powerpc/platforms/85xx/p3041_ds.c b/arch/powerpc/platforms/85xx/p3041_ds.c
index 0ed52e1..089623a 100644
--- a/arch/powerpc/platforms/85xx/p3041_ds.c
+++ b/arch/powerpc/platforms/85xx/p3041_ds.c
@@ -43,6 +43,18 @@ static int __init p3041_ds_probe(void)
 	return of_flat_dt_is_compatible(root, "fsl,P3041DS");
 }
 
+#if defined(CONFIG_PHYLIB) && defined(CONFIG_VITESSE_PHY)
+int vsc824x_add_skew(struct phy_device *phydev);
+#define PHY_ID_VSC8244                  0x000fc6c0
+static int __init board_fixups(void)
+{
+	phy_register_fixup_for_uid(PHY_ID_VSC8244, 0xfffff, vsc824x_add_skew);
+
+	return 0;
+}
+machine_device_initcall(p3041_ds, board_fixups);
+#endif
+
 define_machine(p3041_ds) {
 	.name			= "P3041 DS",
 	.probe			= p3041_ds_probe,
@@ -55,9 +67,12 @@ define_machine(p3041_ds) {
 	.restart		= fsl_rstcr_restart,
 	.calibrate_decr		= generic_calibrate_decr,
 	.progress		= udbg_progress,
+	.idle_loop		= cpu_idle_simple,
+	.power_save		= ppc_wait,
+	.init_early		= corenet_ds_init_early,
 };
 
-machine_device_initcall(p3041_ds, corenet_ds_publish_devices);
+machine_device_initcall(p3041_ds, declare_of_platform_devices);
 
 #ifdef CONFIG_SWIOTLB
 machine_arch_initcall(p3041_ds, swiotlb_setup_bus_notifier);
-- 
1.7.0.4

